base/checked.jl

```diff
@@ -90,16 +90,18 @@ The overflow protection may impose a perceptible performance penalty.
 function checked_neg(x::T) where T<:Integer
     checked_sub(T(0), x)
 end
+throw_overflowerr_negation(x) = (@_noinline_meta;
+    throw(OverflowError("cannot compute -x for x = $x::$(typeof(x))")))
```

Comment:
While you're at it, it would be useful to mention "checked" somewhere, so that it's clear this is not an error from the standard `-` function. Same for `abs`.